### ISSUE TITLE
kokkos: `matrix_vector_product_lower` test failing

### ISSUE THREAD

**Issue by MikolajZuzek** [kokkos/stdBLAS#223]
@mhoemmen @fnrizzi 

`utest_{overwriting,updating}_{symmetric,hermitian}_matrix_vector_product_lower` tests failed on [CI job](https://github.com/kokkos/stdBLAS/runs/6607891533?check_suite_focus=true#step:11:732):
```shell
38/69 Test #38: utest_overwriting_symmetric_matrix_vector_product_lower ....***Failed  Error regular expression found in output. Regex=[[^a-z]Error]  0.02 sec
-- Found UnixCommands: /usr/bin/bash  
numerical test succeeded
CMake Error at /home/runner/work/stdBLAS/stdBLAS/src/stdblas/tests/kokkos-based/test.cmake:24 (message):
  test failed: overwriting_symmetric_matrix_vector_product did not call the
  correct Kokkos impl
```
and then on [the re-run of the same job](https://github.com/kokkos/stdBLAS/runs/6608095520?check_suite_focus=true) they passed fine, which suggests this failure happens randomly...

**Comment by fnrizzi** [comment]
Are you able to reproduce on your machine? Or just the CI fails?

**Comment by MikolajZuzek** [comment]
@fnrizzi
Yes, I reproduced this locally and it happens for both `lower` and `upper`: the easiest way to reproduce is to execute the tests in parallel, like:
```shell
ctest -j 8 -R "(lower|upper)"
```
Please see the fix proposed on #228.
